src/core.jl

```diff
@@ -156,12 +202,13 @@ corresponding colors from `[color]` which defaults to `oneunit(eltype(img))`
 If only a single color `color` is specified then all objects will be
 colored with that color.
 """
-function draw!(img::AbstractArray{T,2}, objects::AbstractVector{U}, colors::AbstractVector{V}) where {T<:Colorant, U<:Drawable, V<:Colorant}
+function draw!(img::AbstractArray{T,2}, objects::AbstractVector{U}, colors::AbstractVector{V}; in_bounds::AbstractVector{Bool}=[false], thickness::AbstractVector{<:Integer}=[-1]) where {T<:Colorant, U<:Drawable, V<:Colorant}
     colors = copy(colors)
-    while length(colors) < length(objects)
-        push!(colors, oneunit(T))
-    end
-    foreach((object, color) -> draw!(img, object, color), objects, colors)
+    while length(colors) < length(objects) push!(colors, oneunit(T)) end
```

Comment:
> If only a single color `color` is specified then all objects will be
colored with that color.	colored with that color.

The current codes don't represent this logic well.

```julia
function draw!(img::AbstractArray{T, 2}, objects::AbstractVector{U}, color::Colorant; in_bounds::Bool=false, thickness::Integer=-1) where {T<:Colorant, U<:Drawable}
    draw!(img, objects, repeat([color], length(objects));
          in_bounds = repeat([in_bounds], length(objects)), 
          thinckness = repeat([thickness], length(objects)))
end
function draw!(img::AbstractArray{T,2}, objects::AbstractVector{U}, colors::AbstractVector{V}; in_bounds::AbstractVector{Bool}, thickness::AbstractVector{<:Integer}) where {T<:Colorant, U<:Drawable, V<:Colorant}
    length(colors) == length(objects) || throw("The number of colors and objects should be equal.")
    length(in_bounds) == length(objects) || throw("")
    length(thickness) == length(objects) || throw("")
    foreach((object, color, in_b, thick) -> draw!(img, object, color), objects, colors, in_bounds, thickness)
    img
end
```